diff --git a/spec/ruby/library/win32ole/win32ole/ole_func_methods_spec.rb b/spec/ruby/library/win32ole/win32ole/ole_func_methods_spec.rb
index b1a52bc78c..67639b9069 100644
--- a/spec/ruby/library/win32ole/win32ole/ole_func_methods_spec.rb
+++ b/spec/ruby/library/win32ole/win32ole/ole_func_methods_spec.rb
@@ -14,7 +14,14 @@
     end
 
     it "raises ArgumentError if argument is given" do
-      lambda { @ie.ole_func_methods(1) }.should raise_error ArgumentError
+      err = nil
+      begin
+        @ie.ole_func_methods(1)
+      rescue => e
+        err = e
+      ensure
+        err.class.should == ArgumentError
+      end
     end
 
     it "returns an array of WIN32OLE_METHODs" do
